src/scripts.ts

```diff
@@ -47,14 +46,17 @@ export class ScriptLinter extends EventEmitter {
   };
 
   private lint = (files: string[]) => {
-    const lintOptions = {configuration: this.configuration, formatter: "verbose"};
+    const lintOptions: ILinterOptions = {fix: false, formatter: "verbose"};
 
     return Promise.all(files.map(file =>
       utils.readFile(file).then(content => ({file, content}))))
       .then((fileObjs) =>
         new Promise((resolve, reject) => {
           for (let fileObj of fileObjs) {
-            const lintingResult = new Linter(fileObj.file, fileObj.content, lintOptions).lint();
+            const linter = new Linter(lintOptions);
```

Comment:
Hm... it's possible to instantiate the linter only once, yes. However, if we do so, we get a single linter result for **all** linted files, which makes it more complicated to re-map the result to properly adjust our `lintingErrors` object. The code would roughly look like:
```typescript
const linter = new Linter(lintOptions);
for (let fileObj of fileObjs) {
  linter.lint(fileObj.file, fileObj.content, this.configuration);
}
const lintingResult = linter.getResult();
const perFile = _.groupBy(lintingResult.failures, (failure: RuleFailure) => failure.getFileName());
_.each(perFile, (failures: RuleFailure[], fileName: string) => {
  if (failures.length === 0) {
    delete this.lintingErrors[fileName];
  } else {
    this.lintingErrors[fileName] = failures;
  }
});
```
The problem is that we have to group `lintingResult.failures` after retrieving all failures, but files that do not contain any more failures are no longer on that list.

To properly deal with this, it would be required to also add entries from `lintingErrors` to `perFile`, and handle them as empty arrays...

Long story short: I'd suggest using a single instance of the linter makes the code more complicated than it is atm., while only having a minor potential to save runtime performance (I'd even guess it would worse).